li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -137,7 +144,30 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener callb
 
   @Override
   public void assign(Collection<TopicPartition> partitions) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (partitions == null) {
+      throw new IllegalArgumentException("Topic partition collection to assign to cannot be null");
+    }
+
+    // If partitions are empty, it should be treated the same as unsubscribe().
+    if (partitions.isEmpty()) {
```

Comment:
update the comment to reflect that this is the vanilla client behaviour (as its a weird one)

Reply:
Done